bigquery_etl/docs/validate_docs.py

```diff
@@ -67,12 +38,21 @@ def validate(project_dirs):
             for root, dirs, files in os.walk(project_dir):
                 if os.path.basename(root) == EXAMPLE_DIR:
                     for file in files:
-                        dry_run_sql = sql_for_dry_run(
-                            os.path.join(root, file), parsed_routines, project_dir
+                        dry_run_sql = sub_local_routines(
+                            (Path(root) / file).read_text(),
+                            project_dir,
+                            parsed_routines,
                         )
 
                         # store sql in temporary file for dry_run
-                        tmp_dir = Path(tempfile.mkdtemp()) / Path(root)
+                        # DryRun will use tmp_example_file.parent.parent.name as the
+                        # default dataset, which must exist in moz-fx-data-shared-prod
+                        tmp_dir = (
+                            Path(tempfile.mkdtemp())
+                            / "moz-fx-data-shared-prod"
+                            / "tmp"
+                            / "example"
+                        )
```

Comment:
this fixes an issue where udf examples would have a default dataset in the wrong project. For example `mozfun.norm.fenix_app_info`, but `moz-fx-data-shared-prod:norm` does not exist because the default dataset should be `mozfun:norm`. DryRun doesn't support default datasets in a different project, so instead this overrides the default dataset to be `tmp`